js/store.js

```diff
@@ -113,11 +203,49 @@ const store = {
 
     this._write(websites);
 
-    if (newSite) {
+    if (newSite && data['isVisible']) {
```

Comment:
If the previous statement is true about this key not always being present this would be a bug here.